src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,197 @@
+  const visibleContexts = node.contexts.filter((c) => c.isVisible && !c.rootId.startsWith("urn:programme"));
+  // A resource can have several context in one subject. rootId and parentId helps picking the right one.
+  const rootContexts = rootId ? visibleContexts.filter((c) => c.rootId === rootId) : visibleContexts;
+  const parentContexts = parentId ? rootContexts.filter((c) => c.parentIds.includes(parentId)) : rootContexts;
+
+  const selectedCtx = parentContexts?.[0];
+  const path = selectedCtx?.path || node.path;
+  const rank = selectedCtx?.rank;
+  const contextId = selectedCtx?.contextId;
+  const relevanceId = selectedCtx?.relevanceId;
+  const entity = nodeToTaxonomyEntity({ ...node, contexts: visibleContexts }, language);
+  return { ...entity, contextId, path, rank, relevanceId };
+};
+
+export const Query = {
+  async node(
+    _: any,
+    { id, rootId, parentId, contextId }: GQLQueryNodeArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | undefined> {
+    if (id) {
+      const node = await context.loaders.nodeLoader.load({ id, rootId, parentId });
+      return pickContextValues(node, context.language, rootId, parentId);
+    }
+    if (contextId) {
+      const nodes = await queryNodes({ contextId, includeContexts: true, filterProgrammes: true }, context);
+      if (nodes.length === 0) {
+        throw new Error(`No node found with contextId: ${contextId}`);
+      }
+      const entities = nodes.map((node) => nodeToTaxonomyEntity(node, context.language, contextId));
+      return entities[0];
+    }
+    throw new Error("Missing id or contextId");
+  },
+  async nodes(
+    _: any,
+    { nodeType, contentUri, filterVisible, metadataFilterKey, metadataFilterValue, ids }: GQLQueryNodesArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode[]> {
+    const params = {
+      ids: ids,
+      includeContexts: true,
+      filterProgrammes: true,
+      language: context.language,
+    };
+    const nodes: Node[] = [];
+    if (contentUri) {
+      nodes.concat(await queryNodes({ contentURI: contentUri, ...params }, context));
+    } else if (nodeType) {
+      nodes.concat(await queryNodes({ nodeType: nodeType, ...params }, context));
+    } else if (metadataFilterKey) {
+      nodes.concat(await queryNodes({ key: metadataFilterKey, value: metadataFilterValue, ...params }, context));
+    }
+    const filtered = filterVisible ? nodes.filter((node) => node.contexts.find((context) => context.isVisible)) : nodes;
+    return filtered.map((node) => {
+      return nodeToTaxonomyEntity(node, context.language);
+    });
+  },
+  async nodeByArticleId(
+    _: any,
+    { articleId, nodeId }: GQLQueryNodeByArticleIdArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | null> {
+    let node = null;
+    if (articleId) {
+      node = await fetchNodeByContentUri(`urn:article:${articleId}`, context);
+    }
+    if (nodeId) {
+      node = await fetchNode({ id: nodeId }, context);
+    }
```

Comment:
Hva skjer hvis begge er definert her?

Reply:
Då hentes det jammen to ganger. Det kan eg sikkert endre på.